telegram/ext/_picklepersistence.py

```diff
@@ -323,12 +315,11 @@ def update_user_data(self, user_id: int, data: UD) -> None:
 
         Args:
             user_id (:obj:`int`): The user the data might have been changed for.
-            data (:obj:`dict` | :attr:`telegram.ext.ContextTypes.user_data`): The
-                :attr:`telegram.ext.Dispatcher.user_data` ``[user_id]``.
+            data (:obj:`dict`): The :attr:`telegram.ext.Dispatcher.user_data` ``[user_id]``.
         """
         if self.user_data is None:
-            self.user_data = defaultdict(self.context_types.user_data)
-        if self.user_data.get(user_id) == data:
+            self.user_data = {}
+        if self.user_data.get(user_id, None) == data:  # type: ignore[arg-type]
```

Comment:
```suggestion
        if self.user_data.get(user_id) == data:  # type: ignore[arg-type]
```
what's the `type ignore` for?

Reply:
When we add `None`, it complains that the default arg doesn't match type of `user_data` (`UD`). Anyway this is fixed in latest mypy version, which we probably should upgrade to soon

Reply:
ah okay. feel free to upgrade, then. still, the `None` should not be needed 🤔

Reply:
> ah okay. feel free to upgrade, then.

we'll do that in another PR, there's false positives in constants module + other things